### PR TITLE
Initial attempt to disable IDs

### PR DESCRIPTION
**Pull request by ttomdewit**
Initial attempt to disable IDs which closes https://github.com/laravel-shift/blueprint/issues/85

### CODE REVIEW

**Comment by ttomdewit** [comment]
- [ ] Add test that verifies ID column is indeed not created

File: tests/fixtures/definitions/without-id.bp [hunk 1]
```diff
@@ -0,0 +1,8 @@
+models:
+  Name:
+    softDeletes
```
Comment (jasonmccreary) [inline tests/fixtures/definitions/without-id.bp:3]: How does this test `id: false`?

Comment (ttomdewit) [reply tests/fixtures/definitions/without-id.bp:3]: Ah yes. I copied the original test and stub over to the new test and needed some attributes to remain. Figured I’d only delete the ID attribute.

Comment (jasonmccreary) [reply tests/fixtures/definitions/without-id.bp:3]: I guess I don't understand what this test proves. I thought you were trying to support `id: false` disabling the id field. As such, I'd expect this definition to have `id: false` or the test to fail...

File: tests/Feature/Lexers/ModelLexerTest.php [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 use Blueprint\Lexers\ModelLexer;
 use Tests\TestCase;
+use function var_dump;
```
Comment (jasonmccreary) [inline tests/Feature/Lexers/ModelLexerTest.php:7]: Please remove unused function imports.

File: src/Generators/MigrationGenerator.php [hunk 3]
```diff
@@ -69,9 +69,9 @@ protected function buildDefinition(Model $model)
         /** @var \Blueprint\Models\Column $column */
         foreach ($model->columns() as $column) {
             $dataType = $column->dataType();
-            if ($column->name() === 'id' && $dataType === 'id') {
+            if ($column->name() === 'id' && $dataType === 'id' && $model->usesPrimaryKey()) {
                 $dataType = 'bigIncrements';
-            } elseif ($dataType === 'id') {
+            } elseif ($dataType === 'id' && $model->usesPrimaryKey()) {
```
Comment (jasonmccreary) [inline src/Generators/MigrationGenerator.php:74]: Why does this line need the check? Shouldn't it only be for disabling the `id` column?

File: tests/fixtures/definitions/without-id.bp (hunk 1 above)
Comment (nathane) [reply tests/fixtures/definitions/without-id.bp:3]: Omitting `id` should not change the current behavior. Explicitly setting`id: false` should.

Comment (jasonmccreary) [reply tests/fixtures/definitions/without-id.bp:3]: Exactly. That's why I expect this fixture to have `id: false` based on the test.

Comment (ttomdewit) [reply tests/fixtures/definitions/without-id.bp:3]: So sorry. I’ll fix this!

Comment (ttomdewit) [reply tests/fixtures/definitions/without-id.bp:3]: I explicitly set `id: false` and updated the test to prove `'id' => false` is set.

File: src/Generators/MigrationGenerator.php (hunk 3 above)
Comment (ttomdewit) [reply src/Generators/MigrationGenerator.php]: I misunderstood this line in the code.

**Comment by ttomdewit** [comment]
Thank you

**Comment by jasonmccreary** [comment]
Sure. I did [some refactoring](https://github.com/laravel-shift/blueprint/commit/0ff38dd72ef9ca8b3e6cc3f423b729a7f691a78c) after merging if you want to take a look.

**Comment by ttomdewit** [comment]
> Sure. I did [some refactoring](https://github.com/laravel-shift/blueprint/commit/0ff38dd72ef9ca8b3e6cc3f423b729a7f691a78c) after merging if you want to take a look.

I saw that. Will use that when I attempt to add UUID as shorthand later tonight (no promises, but I'm going to try).

**Pull request merged**